dlib/image_transforms/interpolation.h

```diff
@@ -229,21 +229,22 @@ namespace dlib
     class interpolate_bilinear
     {
     public:
-
         template <typename T, typename image_view_type, typename pixel_type>
-        typename disable_if<is_rgb_image<image_view_type>,bool>::type operator() (
+        bool operator() (
             const image_view_type& img,
-            const dlib::vector<T,2>& p,
+            const dlib::vector<T, 2>& p,
             pixel_type& result
         ) const
         {
-            COMPILE_TIME_ASSERT(pixel_traits<typename image_view_type::pixel_type>::has_alpha == false);
+            // Interpolation currently supports only fully cartesian (non-polar) spaces.
+            COMPILE_TIME_ASSERT(is_color_space_cartesian_image<image_view_type>::value == true);
```

Comment:
This would be better as a `static_assert`.  It wasn't until about 2017 I think, or 2018 that dlib could use c++11, since there are a bunch of dlib users that were stuck on old compilers.  It also took until about that time for visual studio to have C++11 support.  So `COMPILE_TIME_ASSERT` was a workaround for not having `static_assert`.  But C++11 can be used freely in dlib now so no reason not use use `static_assert` here.